app/src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest.kt

```diff
@@ -528,6 +542,31 @@ class PinPasswordActivityTest {
     }
   }
 
+  fun getCurrentActivity(): Activity? {
```

Comment:
Make sure you always solve all the possible mentioned warnings/suggestions by android studio.
```
private fun getCurrentActivity(): Activity? {
    var currentActivity: Activity? = null
    getInstrumentation().runOnMainSync {
      run {
        currentActivity = ActivityLifecycleMonitorRegistry.getInstance().getActivitiesInStage(
          Stage.RESUMED
        ).elementAtOrNull(0)
      }
    }
    return currentActivity
  }

  private inline fun <reified T : Activity> isVisible(): Boolean {
    val am =
      InstrumentationRegistry.getInstrumentation().targetContext.getSystemService(ACTIVITY_SERVICE) as ActivityManager
    val visibleActivityName = this.getCurrentActivity()!!::class.java.name
    return visibleActivityName == T::class.java.name
  }

  private inline fun <reified T : Activity> waitUntilActivityVisible() {
    val startTime = System.currentTimeMillis()
    while (!isVisible<T>()) {
      Thread.sleep(CONDITION_CHECK_INTERVAL)
      if (System.currentTimeMillis() - startTime >= TIMEOUT) {
        throw AssertionError("Activity ${T::class.java.simpleName} not visible after $TIMEOUT milliseconds")
      }
    }
  }
```


Also, you can create top level constants like this:
```
private const val TIMEOUT = 1000L
private const val CONDITION_CHECK_INTERVAL = 100L

@RunWith(AndroidJUnit4::class)
class PinPasswordActivityTest {
```